lib/ansible/modules/cloud/amazon/cloudwatchlogs_log_group_metric_filter.py

```diff
@@ -0,0 +1,191 @@
+    filter_name: BoxFreeStorage
+    filter_pattern: '{ ($.value = *) && ($.hostname = "box")}'
+    state: present
+    metric_transformation:
+        metricName: box_free_space
+        metricNamespace: fluentd_metrics
+        metricValue: "$.value"
+
+- name: delete metric filter on log group /fluentd/testcase
+    lekker_cloudwatch_metric_filter:
+    log_group_name: /fluentd/testcase
+    filter_name: BoxFreeStorage
+    state: absent
+'''
+
+RETURN = """
+logs:
+    description: Return the origin response value
+    returned: success
+    type: list
+    contains:
+        creationTime:
+        filterName:
+        filterPattern:
+        logGroupName:
+        metricTransformation:
+"""
+from ansible.module_utils.aws.core import AnsibleAWSModule, is_boto3_error_code, get_boto3_client_method_parameters
+
+try:
+    from botocore.exceptions import ClientError, BotoCoreError, WaiterError
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+
+def metricTransformationHandler(metricTransformations, originMetricTransformations=None):
+
+    if originMetricTransformations:
+        change = False
+        for item in ["defaultValue", "metricName", "metricNamespace", "metricValue"]:
+            if metricTransformations.get(item) != originMetricTransformations.get(item):
+                change = True
+    else:
+        change = True
+
+    if metricTransformations.get("defaultValue"):
+        retval = [
+            {
+                'metricName': metricTransformations.get("metricName"),
+                'metricNamespace': metricTransformations.get("metricNamespace"),
+                'metricValue': metricTransformations.get("metricValue"),
+                'defaultValue': metricTransformations.get("defaultValue")
+            }
+        ]
+    else:
+        retval = [
+            {
+                'metricName': metricTransformations.get("metricName"),
+                'metricNamespace': metricTransformations.get("metricNamespace"),
+                'metricValue': metricTransformations.get("metricValue"),
+            }
+        ]
+
+    return retval, change
+
+
+def main():
+
+    arg_spec = dict(
+        state=dict(choices=['present', 'absent'], default='present'),
+        log_group_name=dict(type='str', required=True),
+        filter_name=dict(type='str', required=True),
+        filter_pattern=dict(type='str'),
+        metric_transformation=dict(
+            type='dict', default=dict()),
+    )
+
+    module = AnsibleAWSModule(
+        argument_spec=arg_spec,
+        supports_check_mode=True
+    )
+
+    log_group_name = module.params.get("log_group_name")
+    filter_name = module.params.get("filter_name")
+    filter_pattern = module.params.get("filter_pattern")
+    metric_transformation = module.params.get("metric_transformation")
+    state = module.params.get("state")
+
+    cwl = module.client('logs')
+
+    # check if metric filter exists
+    response = cwl.describe_metric_filters(
```

Comment:
You might need to use a paginator here if there might ever be more than 50 results

Reply:
No, because this module sets a metric filter for `LogGroupName && filterName`. 

With the combination of

```
response = cwl.describe_metric_filters(
        logGroupName="/ecs/access,
        filterNamePrefix="404"
    )
```

is only one match possible.  
Either there is already one filter and we fetch the values of the metric filter, or we create the metric filter.